packages/ui-components/src/components/layout/Dialog.vue

```diff
@@ -164,4 +170,11 @@ const onClose = () => {
   if (props.preventCloseOnClickOutside) return
   open.value = false
 }
+
+const onScroll = (e: Event) => {
```

Comment:
You should probably throttle this. There's a lodash utility for that

Reply:
`const onScroll = throttle((e: Event) => {
  const target = e.target as HTMLElement
  const { scrollTop, offsetHeight, scrollHeight } = target
  scrolledFromTop.value = scrollTop > 0
  scrolledToBottom.value = scrollTop + offsetHeight >= scrollHeight
}, 60)`

60ms is the highest I could go without it being too delayed (and looking like a bug) when you reach the top or bottom.